From df0e7c1414b793a36e34a5875e4196a3a45704a5 Mon Sep 17 00:00:00 2001
From: Kuiying Wang <kuiying.wang@intel.com>
Date: Mon, 21 Dec 2020 23:09:33 +0800
Subject: [PATCH] Fix crash issue due to throw undefined error

Below 3 error is used by bios config manager daemon.
xyz.openbmc_project.Common.Error.InvalidArgument
xyz.openbmc_project.BIOSConfig.Common.Error.AttributeNotFound
xyz.openbmc_project.BIOSConfig.Common.Error.AttributeReadOnl

Tested:
bios config manager daemon could throw these 3 error,
but not crash any more.

Signed-off-by: Kuiying Wang <kuiying.wang@intel.com>
---
 xyz/openbmc_project/BIOSConfig/Manager.interface.yaml | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/xyz/openbmc_project/BIOSConfig/Manager.interface.yaml b/xyz/openbmc_project/BIOSConfig/Manager.interface.yaml
index 225ccaa..98daf3a 100644
--- a/xyz/openbmc_project/BIOSConfig/Manager.interface.yaml
+++ b/xyz/openbmc_project/BIOSConfig/Manager.interface.yaml
@@ -109,7 +109,10 @@ properties:
               ex- { {"QuietBoot",Type.Integer, 0x1},
                     { "DdrFreqLimit",Type.String,"2933"}
                   }
-
+      errors:
+        - xyz.openbmc_project.Common.Error.InvalidArgument
+        - xyz.openbmc_project.BIOSConfig.Common.Error.AttributeNotFound
+        - xyz.openbmc_project.BIOSConfig.Common.Error.AttributeReadOnly
 
 enumerations:
     - name: AttributeType
-- 
2.17.1

